### ISSUE TITLE
Human readable name not used in parameter error messages

### ISSUE THREAD

**Issue by gar1t** [pallets/click#855]
I'm getting error messages that look like this:

```
Usage: guild info [OPTIONS] PACKAGE
Try "guild info --help" for help.

Error: Missing argument "package".
```

IMO the last line should read `Missing argument PACKAGES` - "packages" is not reflected in the usage.

I believe the issue is here (and similar patterns in exceptions.py):

https://github.com/pallets/click/blob/master/click/exceptions.py#L121

where `self.param.opts` short-circuits the use of the human readable name, which would use metavar as expected.

**Comment by cdunklau** [comment]
This looks like a dupe of #598

**Comment by LokiHokie** [comment]
Fixed #709
